dcc-submission-server/src/main/java/org/icgc/dcc/submission/server/service/ReleaseService.java

```diff
@@ -613,18 +577,25 @@ public void removeQueuedSubmissions(@NonNull String... targets) throws InvalidSt
       }
     }
 
-    releaseRepository.updateRelease(releaseName, release);
+    releaseRepository.updateReleaseQueue(releaseName, release.getQueue());
+    submissionService.updateExistingSubmissions(submissions.values());
   }
 
   public void resetSubmissions() {
-    resetSubmissions(getNextRelease().getProjectKeys());
+    val projectKeys = submissionService.findReleaseProjectKeys(getNextRelease().getName());
+    resetSubmissions(projectKeys);
   }
 
   @Synchronized
   public void resetInvalidSubmissions() {
     val release = getNextRelease();
     val filePatternToTypeMap = dictionaryRepository.getFilePatternToTypeMap(release.getDictionaryVersion());
-    for (val projectKey : release.getInvalidProjectKeys()) {
+    val invalidProjectKeys = submissionService.findSubmissionsByReleaseName(release.getName()).stream()
```

Comment:
This is very expensive (submissions are big!). Lets push down the predicate into the query.